bot/decorators.py

```diff
@@ -46,3 +49,24 @@ def in_channel(channel_id):
                   f"The result of the in_channel check was {check}.")
         return check
     return commands.check(predicate)
+
+
+def locked():
+    """
+    Allows the user to only run one instance of the decorated command at a time.
+    Subsequent calls to the command from the same author are
+    ignored until the command has completed invocation.
+    This decorator has to go before (below) the `command` decorator.
+    """
+    def wrap(func):
+        func.__locks = WeakValueDictionary()
+
+        @wraps(func)
+        async def inner(self, ctx, *args, **kwargs):
+            lock = func.__locks.setdefault(ctx.author.id, Lock())
+            if lock.locked():
+                return
+            async with func.__locks.setdefault(ctx.author.id, Lock()):
```

Comment:
`func.__locks` is a `WeakValueDictionary`, and I'm not aware of `WeakValueDictionary.__aenter__ ` being a thing. Am I missing something?

Reply:
this question is better directed at Ava, I have no idea what this code is doing and just blindly implemented it because Ava tends to know what she's doing.

Reply:
Also it might be nice to return something to the user saying the command is already running